flik/linesearch/conditions.py

```diff
@@ -1 +1,47 @@
 """Conditions for the line search algorithm."""
+def soft_wolfe(gradf, valf, alpha, direction, const2=0.9):
+    """Curvature condition
+
+    Parameters
+    ----------
+    gradf: function
+        Gradient of objective nonlinear function
+    valf: np.ndarray(N,)
+        Function value
+    alpha: float
+        Step length
+    direction: np.ndarray(N,)
+        Descent direction
+    const2: float
+        Constant C2
+
+    Raises
```

Comment:
I've never thought of adding what errors raises, but I guess we should be consistent, so I'll add something similar then